main.go

```diff
@@ -106,11 +107,24 @@ func main() {
 	http.Handle("/live", handlers.NewLivenessHandler())
 	http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
 
-	log.Info("Starting http handler...")
-	address := fmt.Sprintf(":%s", Options.ListenPort)
 	if Options.HTTPSKeyFile != "" && Options.HTTPSCertFile != "" {
-		log.Fatal(http.ListenAndServeTLS(address, Options.HTTPSCertFile, Options.HTTPSKeyFile, nil))
+		if Options.PlainHTTPListenPort != "" {
+			log.Infof("Starting https handler on %s...", Options.ListenPort)
+			go func() {
+				log.Fatal(http.ListenAndServeTLS(fmt.Sprintf(":%s", Options.ListenPort), Options.HTTPSCertFile, Options.HTTPSKeyFile, nil))
+			}()
+
+			log.Infof("Starting http handler on %s...", Options.PlainHTTPListenPort)
+			go func() {
+				log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", Options.PlainHTTPListenPort), nil))
+			}()
+			select {}
+		} else {
+			log.Infof("Starting https handler on %s...", Options.ListenPort)
+			log.Fatal(http.ListenAndServeTLS(fmt.Sprintf(":%s", Options.ListenPort), Options.HTTPSCertFile, Options.HTTPSKeyFile, nil))
+		}
 	} else {
-		log.Fatal(http.ListenAndServe(address, nil))
+		log.Infof("Starting http handler on %s...", Options.ListenPort)
+		log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", Options.ListenPort), nil))
```

Comment:
Reworkd this:
* define `httpListen` and `httpsListen` so that these can be reused (`httpsListen` is used once, but its easier to keep it uniform)
* in https mode always run `httpsListen` as a goroutine. If `HTTPListenPort` is set we start http listerner as goroutine - and wait for it to be completed in both cases